.github/workflows/test.yml

```diff
@@ -31,9 +36,9 @@ jobs:
         uses: elastic/apm-pipeline-library/.github/actions/version-framework@current
         with:
           # Use .ci/.matrix_python_full.yml if it's a scheduled workflow, otherwise use .ci/.matrix_python.yml
-          versionsFile: .ci/.matrix_python${{ github.event_name == 'schedule' && '_full' || '' }}.yml
+          versionsFile: .ci/.matrix_python${{ (github.event_name == 'schedule' || github.event_name == 'push' || inputs.full-matrix) && '_full' || '' }}.yml
           # Use .ci/.matrix_framework_full.yml if it's a scheduled workflow, otherwise use .ci/.matrix_framework.yml
-          frameworksFile: .ci/.matrix_framework${{ github.event_name == 'schedule' && '_full' || '' }}.yml
+          frameworksFile: .ci/.matrix_framework${{ (github.event_name == 'schedule' || github.event_name == 'push' || inputs.full-matrix) && '_full' || '' }}.yml
```

Comment:
Will this happen for push events on any other upstream branch but `main`?  I don't know how often those branches are created, but I wanted to note it.

Reply:
The trigger is configured to `main`. Hence, it will only happen for the main branch.

```yaml
  push:
    branches:
      - main
    paths-ignore:
      - "**/*.md"
      - "**/*.asciidoc"
```